flight/Modules/OnScreenDisplay/onscreendisplay.c

```diff
@@ -0,0 +1,1448 @@
+{
+	osd_page_updated = true;
+}
+
+void update_module_settings(UAVObjEvent *ev)
+{
+	ModuleSettingsAdminStateGet(module_state);
+}
+
+/**
+ * Start the osd module
+ */
+int32_t OnScreenDisplayStart(void)
+{
+	if (module_enabled)
+	{
+		onScreenDisplaySemaphore = PIOS_Semaphore_Create();
+
+		taskHandle = PIOS_Thread_Create(onScreenDisplayTask, "OnScreenDisplay", STACK_SIZE_BYTES, NULL, TASK_PRIORITY);
+		TaskMonitorAdd(TASKINFO_RUNNING_ONSCREENDISPLAY, taskHandle);
+
+		ModuleSettingsConnectCallback(update_module_settings);
+
+#if defined(PIOS_INCLUDE_WDG) && defined(OSD_USE_WDG)
+		#error HERE
+		PIOS_WDG_RegisterFlag(PIOS_WDG_OSDGEN);
+#endif
+		return 0;
+	}
+	return -1;
+}
+
+
+/**
+ * Initialize the osd module
+ */
+int32_t OnScreenDisplayInitialize(void)
+{
+	uint8_t osd_state;
+
+	ModuleSettingsAdminStateGet(module_state);
+
+	OnScreenDisplayPageSettingsInitialize();
+	OnScreenDisplayPageSettings2Initialize();
+	OnScreenDisplayPageSettings3Initialize();
+	OnScreenDisplayPageSettings4Initialize();
+
+	// initialize other objects that we access
+	AccessoryDesiredInitialize();
+	AirspeedActualInitialize();
+	AttitudeActualInitialize();
+	BaroAltitudeInitialize();
+	FlightStatusInitialize();
+	FlightBatteryStateInitialize();
+	GPSPositionInitialize();
+	PositionActualInitialize();
+	GPSTimeInitialize();
+	GPSSatellitesInitialize();
+	GPSVelocityInitialize();
+	HomeLocationInitialize();
+	ManualControlCommandInitialize();
+	SystemAlarmsInitialize();
+	SystemStatsInitialize();
+	TabletInfoInitialize();
+	TaskInfoInitialize();
+	VelocityActualInitialize();
+	WaypointInitialize();
+	WaypointActiveInitialize();
+
+	OnScreenDisplaySettingsOSDEnabledGet(&osd_state);
+	if (osd_state == ONSCREENDISPLAYSETTINGS_OSDENABLED_ENABLED) {
+		module_enabled = true;
+	} else {
+		module_enabled = false;
+		return 0;
+	}
+
+	PIOS_Video_Init(&pios_video_cfg);
+
+	/* Register callbacks for modified settings */
+	OnScreenDisplaySettingsConnectCallback(OnScreenSettingsUpdatedCb);
+
+	/* Register callbacks for modified pages */
+	OnScreenDisplayPageSettingsConnectCallback(OnScreenPageUpdatedCb);
+	OnScreenDisplayPageSettings2ConnectCallback(OnScreenPageUpdatedCb);
+	OnScreenDisplayPageSettings3ConnectCallback(OnScreenPageUpdatedCb);
+	OnScreenDisplayPageSettings4ConnectCallback(OnScreenPageUpdatedCb);
+
+	return 0;
+}
+
+
+MODULE_INITCALL(OnScreenDisplayInitialize, OnScreenDisplayStart);
+
+/**
+ * Main osd task. It does not return.
+ */
+#define BLANK_TIME 3000
+#define INTRO_TIME 5000
+static void onScreenDisplayTask(__attribute__((unused)) void *parameters)
+{
+
+	ModuleSettingsAdminStateGet(module_state);
+
+	AccessoryDesiredData accessory;
+	OnScreenDisplaySettingsData osd_settings;
+	OnScreenDisplayPageSettingsData osd_page_settings;
+
+	uint8_t current_page = 0;
+	uint8_t last_page = -1;
+
+	OnScreenDisplaySettingsGet(&osd_settings);
+	home_baro_altitude = 0.;
+
+	// blank
+	while (PIOS_Thread_Systime() <= BLANK_TIME) {
```

Comment:
This needs better documentation. What's the value of this? Is it just to keep the OSD from showing anything before the user has time to put on the googles?